src/System.Net.Http.WinHttpHandler/src/System/Net/Http/WinInetProxyHelper.cs

```diff
@@ -178,5 +181,42 @@ public bool ManualSettingsOnly
 
             return useProxy;
         }
+
+        public bool AutoDetectScriptSuccess()
+        {
+            // Only check WinHttpDetectAutoProxyConfigUrl if auto detect is set.
+            if (!AutoDetect) return false;
+
+            bool doAutoDetect = false;
+            int currentTimeStamp = Environment.TickCount;
+
+            // Environment.TickCount will increment from zero to Int32.MaxValue for approximately 24.9 days, then jump to
+            // Int32.MinValue. When that happens, we may skip calling WinHttpDetectAutoProxyConfigUrl for one request which we
+            // otherwise will call. Since _previousTimeStampTicks will get updated in that request, subsequent requests in
+            // next 24.9 days period will behave correctly.
```

Comment:
My previous comment seems to have disappeared... trying again...

Why would we skip at that point?  e.g. this will output 1:
```C#
int a = int.MaxValue;
int b = int.MinValue;
Console.WriteLine(b - a);
```
I think the only time we'd end up skipping with this logic is if two requests were more than 48 days apart.